packages/nimble-components/src/patterns/dropdown/helpers.ts

```diff
@@ -0,0 +1,24 @@
+import {
+    controlHeight as controlHeightToken,
+    smallPadding
+} from '../../theme-provider/design-tokens';
+
+const maxVisibleOptions = 10.5;
+
+export function getDropdownMaxHeight(
+    element: HTMLElement,
+    filterVisible: boolean
+): number {
+    const controlHeight = parseInt(controlHeightToken.getValueFor(element), 10);
+    const listboxInnerPadding = parseInt(smallPadding.getValueFor(element), 10);
+    const listboxGap = listboxInnerPadding; // both use smallPadding
+    const listboxBorderHeight = 2; // 1px top and bottom
+    const filterHeight = filterVisible ? controlHeight : 0;
+    return (
+        controlHeight * maxVisibleOptions
+        + listboxGap
+        + listboxBorderHeight
+        + filterHeight
+        + listboxInnerPadding
+    );
```

Comment:
@atmgrifter00 @m-akinc Can we keep most of this in CSS? Particularly logic that is taking token values and just adding stuff together. We should be doing the minimal amount of work possible in JS and create private css custom properties that are used in the CSS for the minimal bits of calculation we can't do in CSS. Marking reject on the PR because I don't see any discussion related to keeping layout calculations out of JS so seems overlooked.

Reply:
Yep. Even got to use `:has()` (which should be okay, since it was added to Firefox in Dec 2023)!

Reply:
neat! yea definitely looking much better